provider/aws.go

```diff
@@ -169,20 +200,54 @@ func (p *AWSProvider) submitChanges(zone string, changes []*route53.Change) erro
 		return nil
 	}
 
-	params := &route53.ChangeResourceRecordSetsInput{
-		HostedZoneId: aws.String(expandedHostedZoneID(zone)),
-		ChangeBatch: &route53.ChangeBatch{
-			Changes: changes,
-		},
+	zones, err := p.Zones()
+	if err != nil {
+		return err
 	}
 
-	if _, err := p.Client.ChangeResourceRecordSets(params); err != nil {
-		return err
+	// separate into per-zone change sets to be passed to the API.
+	changesByZone := changesByZone(zones, changes)
+
+	for z, cs := range changesByZone {
+		params := &route53.ChangeResourceRecordSetsInput{
+			HostedZoneId: aws.String(z),
+			ChangeBatch: &route53.ChangeBatch{
+				Changes: cs,
+			},
+		}
+
+		if _, err := p.Client.ChangeResourceRecordSets(params); err != nil {
+			return err
+		}
 	}
 
 	return nil
 }
 
+// changesByZone separates a multi-zone change into a single change per zone.
+func changesByZone(zones map[string]*route53.HostedZone, changeSet []*route53.Change) map[string][]*route53.Change {
+	changes := make(map[string][]*route53.Change)
+
+	for _, z := range zones {
```

Comment:
as discussed an alternative way to do it (and which fixes the duplication for nested hosted zones) would be to iterate over the changes instead of zones, example: 

```
func changesByZone(zones map[string]*route53.HostedZone, changeSet []*route53.Change) map[string][]*route53.Change {
	changes := make(map[string][]*route53.Change)

	for _, c := range changeSet {
                z := findSuitableZone(c, zones)
                if z != nil {
            		changes[aws.StringValue(z.Id)] = append(changes[aws.StringValue(z.Id)], c)
                } else ... //log the error
	}
...
	return changes
}

```